File: php/src/CurrencyService.php
<?php

namespace Aardling\Currency;

class CurrencyService
{
    private $httpClient;

    public function __construct()
    {
        $this->httpClient = new HttpClient();
    }

    public function convert(float $amount, string $baseCurrency, string $targetCurrency): float {
        $value = $this->httpClient->get("https://foreignexchange.example.com/?src=$baseCurrency&tgt=$targetCurrency");
        $newValue = $amount * $value;
        return $newValue;
    }
}

File: java/src/main/java/CurrencyService.java
public class CurrencyService {
    private final HttpClient httpClient;

    public CurrencyService() {
        this.httpClient = new HttpClient();
    }

    public double convert(double amount, String baseCurrency, String targetCurrency) {
        var value = this.httpClient.get("https://foreignexchange.example.com/?src=" + baseCurrency + "&tgt=" + targetCurrency);
        var newValue = amount * value;
        return newValue;
    }
}
